buildscript-utils/src/main/kotlin/com/fueledbycaffeine/spotlight/buildscript/BuildFile.kt

```diff
@@ -46,18 +47,46 @@ internal fun parseBuildFile(
   }
 
   val implicitDependencies = rules
-    .filter { rule ->
+    .flatMap { rule ->
       when (rule) {
-        is BuildscriptMatchRule -> rule.pattern.find(buildscriptContents) != null
-        is ProjectPathMatchRule -> rule.pattern.matches(project.path)
-        is TypeSafeProjectAccessorRule -> true
+        is BuildscriptMatchRule -> {
+          if (rule.pattern.find(buildscriptContents) != null) {
+            rule.includedProjects
+          } else {
+            emptySet()
+          }
+        }
+        is ProjectPathMatchRule -> {
+          if (rule.pattern.matches(project.path)) {
+            rule.includedProjects
+          } else {
+            emptySet()
+          }
+        }
+        is TypeSafeProjectAccessorRule -> {
+          rule.includedProjects
+        }
+        is KotlinGradleScriptNestingRule -> {
+          computeImplicitNestingDependencies(project)
+        }
       }
     }
-    .flatMap { rule -> rule.includedProjects }
 
   return directDependencies + typeSafeProjectDependencies + implicitDependencies
 }
 
+private fun computeImplicitNestingDependencies(project: GradlePath): Set<GradlePath> {
+  // Start with the grandparent directory of the build file
+  // libs/foo/impl/build.gradle.kts -> libs/foo
+  // Then iterate up to the root directory
+  val parentProjectDir = project.buildFilePath.parent.parent
+
+  val rootDir = project.root
+  return generateSequence(parentProjectDir) { it.parent.takeUnless { it == rootDir } }
+    .filter { it.resolve("build.gradle.kts").exists() }
```

Comment:
The behavior of implicitly including parent projects when calling `include(":foo:bar")` is unfortunately a general gradle "feature" and is not specific to kotlin scripting. Square avoids using it as much as possible so I often forget it's a thing.

Reply:
Yup. I have it set up here to look but only actually keep it it if the parent has a kotlin gradle file